crates/volta-core/src/hook/serial.rs

```diff
@@ -1,52 +1,50 @@
+use std::convert::{TryFrom, TryInto};
 use std::marker::PhantomData;
 use std::path::Path;
 
 use super::tool;
-use crate::distro::node::NodeDistro;
-use crate::distro::package::PackageDistro;
-use crate::distro::yarn::YarnDistro;
-use crate::distro::Distro;
 use crate::error::ErrorDetails;
+use crate::tool::{Node, Package, Tool, Yarn};
 use serde::{Deserialize, Serialize};
-use volta_fail::Fallible;
+use volta_fail::{Fallible, VoltaError};
 
 #[derive(Serialize, Deserialize)]
-pub struct ResolveHook {
+pub struct RawResolveHook {
```

Comment:
We actually originally had them all named the same as the internal object (e.g. `serial::ResolveHook` vs `hook::ResolveHook`), but we have an issue #223 to avoid having the same name, as it can get confusing to figure out which one is being referred to. I've been using `Raw` as a prefix when I refactor since that most closely reflects what the types are: The raw result of using `serde`.

I could see renaming the module to `raw`, but that would probably be work for a follow-on PR.

Reply:
That makes sense, and I certainly see the trade off. I think I actually brought it up myself at one point. I’ll keep thinking on the names here. Agreed that it’s be for a follow up regardless!